packages/test/end-to-end-tests/src/test/compatUtils.ts

```diff
@@ -162,6 +168,30 @@ export const createOldRuntimeFactory = (
     return factory as unknown as old.IRuntimeFactory;
 };
 
+function getDataStoreFactory(containerOptions?: ITestContainerConfig) {
+    switch (containerOptions?.fluidDataObjectType) {
+        case undefined:
+        case DataObjectFactoryType.Primed:
+            return createPrimedDataStoreFactory(containerOptions?.registry);
+        case DataObjectFactoryType.Test:
+            return createTestFluidDataStoreFactory(containerOptions?.registry);
+        default:
+            throw new Error("unknown data store factory type");
```

Comment:
Should there be a case here for PrimedWithDependency?

Reply:
Removed PrimedWithDependency :) That should be separate PR, this one just creates ability to much easier add new types.